pkg/reconciler/apiserversource/apiserversource.go

```diff
@@ -115,6 +115,11 @@ func (r *Reconciler) ReconcileKind(ctx context.Context, source *v1.ApiServerSour
 		return err
 	}
 
+	// An empty selector targets all namespaces.
+	// Setting namespaces to an empty slice indicates to the receiver adapter to target all namespaces
+	if isEmptySelector(source.Spec.NamespaceSelector) {
+		namespaces = []string{}
+	}
```

Comment:
I'm not too convinced this is a good way to encode "all namespaces" :thinking:

Reply:
moved to a dedicated config to specify all namespaces

Reply:
cc @evankanderson in can you have opinions here or well-know pattern we could use for this "all namespaces" use case

Reply:
The empty selector meaning all namespaces is something I got from the Sugar controller configuration. And an empty k8s [label selector](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.19/#labelselector-v1-meta) works the same way by targeting all. From the API docs:

```
A label selector is a label query over a set of resources. The result of matchLabels and matchExpressions are ANDed. An empty label selector matches all objects. A null label selector matches no objects.
```

Reply:
Ok, then let's do this